stdlib/Markdown/src/render/terminal/formatting.jl

```diff
@@ -1,68 +1,65 @@
 # This file is a part of Julia. License is MIT: https://julialang.org/license
 
-# Wrapping
+const AnnotIO = Union{AnnotatedIOBuffer, IOContext{AnnotatedIOBuffer}}
 
-function ansi_length(s)
-    replace(s, r"\e\[[0-9]+m" => "") |> textwidth
+function annotprint(f::Function, args...)
+    buf = AnnotatedIOBuffer()
+    f(buf, args...)
+    read(buf, AnnotatedString)
 end
 
-words(s) = split(s, " ")
-lines(s) = split(s, "\n")
+"""
+    with_output_annotations(f::Function, io::AnnotIO, annots::Pair{Symbol, <:Any}...)
 
-function wrapped_line(io::IO, s::AbstractString, width, i)
-    ws = words(s)
-    lines = String[]
-    for word in ws
-        word_length = ansi_length(word)
-        word_length == 0 && continue
-        if isempty(lines) || i + word_length + 1 > width
-            i = word_length
-            if length(lines) > 0
-                last_line = lines[end]
-                maybe_underline = findlast(Base.text_colors[:underline], last_line)
-                if !isnothing(maybe_underline)
-                    # disable underline style at end of line if not already disabled.
-                    maybe_disable_underline = max(
-                        last(something(findlast(Base.disable_text_style[:underline], last_line), -1)),
-                        last(something(findlast(Base.text_colors[:normal], last_line), -1)),
-                    )
+Call `f(io)`, and apply `annots` to the output created by doing so.
+"""
+function with_output_annotations(f::Function, io::AnnotIO, annots::Pair{Symbol, <:Any}...)
+    @nospecialize f annots
+    aio = if io isa AnnotatedIOBuffer io else io.io end
+    start = position(aio) + 1
+    f(io)
+    stop = position(aio)
+    for annot in annots
+        push!(aio.annotations, (start:stop, annot))
+    end
+end
 
-                    if maybe_disable_underline < 0 || maybe_disable_underline < last(maybe_underline)
+"""
+    wraplines(content::AnnotatedString, width::Integer = 80, column::Integer = 0)
 
-                        lines[end] = last_line * Base.disable_text_style[:underline]
-                        word = Base.text_colors[:underline] * word
-                    end
+Wrap `content` into a vector of lines of at most `width` (according to
+`textwidth`), with the first line starting at `column`.
+"""
+function wraplines(content::Annot, width::Integer = 80, column::Integer = 0) where { Annot <: AnnotatedString}
+    s, lines = content.string, SubString{Annot}[]
+    i, lastwrap, slen = firstindex(s), 0, ncodeunits(s)
+    most_recent_break_oppotunity = 1
+    while i < slen
+        if s[i] == ' '
+            most_recent_break_oppotunity = i
+        elseif s[i] == '\n'
+            push!(lines, content[nextind(s, lastwrap):prevind(s, i)])
+            lastwrap = i
+            column = 0
+        elseif column >= width && most_recent_break_oppotunity > 1
+            if lastwrap == most_recent_break_oppotunity
+                nextbreak = findfirst(' ', @view s[nextind(s, lastwrap):end])
```

Comment:
```suggestion
                nextbreak = findfirst(isspace, @view s[nextind(s, lastwrap):end])
```

Reply:
Ah yep, this shouldn't only recognise `' '`.